pkg/tfbridge/x/tokens.go

```diff
@@ -713,27 +747,24 @@ func aliasResource(
 		hist[tfToken].Fields = nil
 	}
 
-	safeRange(res, func(k string, v shim.Schema) {
-		walk(
-			getNonNil(&hist[tfToken].Fields, k),
-			getNonNil(&computed.Fields, k),
-			v,
-		)
-	})
+	walkResource(res, &hist[tfToken].Fields, &computed.Fields)
 }
 
-func getNonNil[K comparable, V any](m *map[K]*V, key K) *V {
+func getNonNil[K comparable, V any](m *map[K]*V, key K) (_ *V, alreadyThere bool) {
 	contract.Assertf(m != nil, "Cannot restore map if ptr is nil")
 	if *m == nil {
 		*m = map[K]*V{}
 	}
 	v := (*m)[key]
+
 	if v == nil {
 		var new V
 		v = &new
 		(*m)[key] = v
+	} else {
+		alreadyThere = true
```

Comment:
Where is `alreadyThere` initialized?

Reply:
As part of the [named return value](https://go.dev/tour/basics/7).

Reply:
I use named return values a bit. They do have a place it seems.